web/src/main/java/com/bazaarvoice/emodb/web/migrator/config/MigratorConfiguration.java

```diff
@@ -0,0 +1,129 @@
+package com.bazaarvoice.emodb.web.migrator.config;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.Optional;
+
+import javax.validation.Valid;
+import javax.validation.constraints.NotNull;
+
+/**
+ * Migrator-specific service configurations.
+ */
+public class MigratorConfiguration {
+
+    private static final int DEFAULT_READ_THREAD_COUNT = 8;
+    private static final int DEFAULT_WRITE_THREAD_COUNT = 5;
+    private static final String DEFAULT_MIGRATE_STATUS_TABLE = "__system_migrate";
+    private static final String DEFAULT_MIGRATE_STATUS_TABLE_PLACEMENT = "app_global:migration";
+    private static final int DEFAULT_MAX_CONCURRENT_WRITES = 500;
+
+    // If using EmoDB queues, the API key to use
+    @Valid
+    @NotNull
+    @JsonProperty ("migrateApiKey")
+    private Optional<String> _migrateApiKey = Optional.absent();
+
+    // Maximum number of read threads that can run concurrently on a single server.  Default is 8.
+    @Valid
+    @NotNull
+    @JsonProperty ("readThreadCount")
+    private int _readThreadCount = DEFAULT_READ_THREAD_COUNT;
+
+    // Maximum number of write threads that can run concurrently on a single server.  Default is 12.
+    @Valid
+    @NotNull
+    @JsonProperty ("writeThreadCount")
+    private int _writeThreadCount = DEFAULT_WRITE_THREAD_COUNT;
```

Comment:
It manages the number of threads writing batches in a particular split. If we remove that logic from the localRangeMigrator, then it will no longer apply